archaius-core/src/main/java/com/netflix/config/PropertyWrapper.java

```diff
@@ -41,12 +41,7 @@
     private final List<Runnable> callbackList = Lists.newArrayList();
 
     static {
-        PropertyWrapper.registerSubClassWithNoCallback(DynamicIntProperty.class);
```

Comment:
So this still incurs the overhead of callbacks both in terms of memory and invocation. Any reason that we don't want to introduce cached properties so people are aware of the cost and can make the choice? I'm concerned people will be surprised by the behavior changes.